src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -309,41 +303,52 @@ public void setPeriod(String period) {
 
     @Override
     public void visitJavadocToken(DetailNode ast) {
-        if (containsSummaryTag(ast)) {
-            validateSummaryTag(ast);
-        }
-        else if (!startsWithInheritDoc(ast)) {
-            final String summaryDoc = getSummarySentence(ast);
-            if (summaryDoc.isEmpty()) {
-                log(ast.getLineNumber(), MSG_SUMMARY_JAVADOC_MISSING);
-            }
-            else if (!period.isEmpty()) {
-                final String firstSentence = getFirstSentence(ast);
-                final int endOfSentence = firstSentence.lastIndexOf(period);
-                if (!summaryDoc.contains(period)) {
-                    log(ast.getLineNumber(), MSG_SUMMARY_FIRST_SENTENCE);
+        final DetailNode inlineSummaryTag = getInlineSummaryTag(ast);
+        if (inlineSummaryTag == null) {
```

Comment:
3 Parter. Part 3, and then just change this to `!Optional.isPresent`.

Reply:
used `isEmpty`, should be fine.